src/plugins/intel_gpu/src/graph/convolution.cpp

```diff
@@ -169,137 +91,70 @@ layout convolution_inst::calc_output_layout(convolution_node const& node, kernel
-                           desc->output_size.spatial[1],
-                           desc->output_size.spatial[2]);
-        if (output_type == data_types::bin) {
-            return {output_type, format::b_fs_yx_32fp, output_size};
-        }
-
-        return {output_type, out_fmt, output_size};
+    if (output_type == data_types::bin) {
+        out_fmt = format::b_fs_yx_32fp;
     }
 
-    auto output_range = calc_sliding_window_output_range<swor_mode::all>(input_layout.get_tensor(),
-                                                                         filter_size,
-                                                                         pad,
-                                                                         stride,
-                                                                         dilation,
-                                                                         true,
-                                                                         1);
-
-    tensor output_size = tensor(input_layout.batch(),
-                                weights_layout.ofm() * weights_layout.group(),
-                                output_range.spatial[0],
-                                output_range.spatial[1],
-                                output_range.spatial[2]);
-
-
-    if (output_type == data_types::bin) {
-        return {output_type, format::b_fs_yx_32fp, output_size};
+    // dynamic case
+    std::vector<ov::PartialShape> input_shapes = {
+        input_layout.get<ov::PartialShape>(),
+        weights_layout.get<ov::PartialShape>()
+    };
+    std::vector<ov::PartialShape> output_shapes;
+
+    auto spatial_rank = input_layout.get_spatial_rank();
+    auto dilation = align_to_spatial_rank(desc->dilation, spatial_rank, static_cast<size_t>(1));
+    auto strides = align_to_spatial_rank(desc->stride, spatial_rank, static_cast<size_t>(1));
+    auto pads_begin = align_to_spatial_rank(desc->padding_begin, spatial_rank, static_cast<std::ptrdiff_t>(0));
+    auto pads_end = align_to_spatial_rank(desc->padding_end, spatial_rank, static_cast<std::ptrdiff_t>(0));
+
+    if (desc->deformable_mode) {
+        ov::op::v8::DeformableConvolution op;
+        op.set_group(desc->groups);
+        op.set_deformable_group(desc->deformable_groups);
+        op.set_dilations(dilation);
+        op.set_strides(strides);
+        op.set_auto_pad(desc->auto_pad);
+        input_shapes.insert(std::next(input_shapes.begin()), impl_param.get_input_layout(1).get_partial_shape());
+        output_shapes = ov::op::v8::shape_infer(&op, input_shapes, pads_begin, pads_end);
+    } else if (desc->grouped_weights_shape || desc->groups > 1) {
+        ov::op::v1::GroupConvolution op;
+        op.set_dilations(dilation);
+        op.set_strides(strides);
+        op.set_auto_pad(desc->auto_pad);
+        auto& weights_shape = input_shapes[1];
+        if (input_shapes[1].size() == 4 && input_shapes[0].size() == 4) {
```

Comment:
That's actually WA for unit tests which create grouped weights w/o explicit group dimension in some cases (which needs to be fixed in the future). Added comment that explains motivation. So it may be applicable for 5d too, but as long as we don't have any failures caused by current version of WA, I'd prefer not to extend it.